lua/lv-formatter/init.lua

```diff
@@ -22,27 +22,38 @@ local function formatter_exists(lang_formatter)
   return true
 end
 
--- returns default formatter for given language
-local function formatter_return(lang_formatter)
-  return {
-    exe = lang_formatter.exe,
-    args = lang_formatter.args,
-    stdin = not (lang_formatter.stdin ~= nil),
-  }
+-- returns default formatter for given language. reads from `O.lang.*.formatter` in default-config.lua
+local function default_formatter(formatter_list)
+  for _, selected_formatter in pairs(formatter_list) do
+    if fmt_conf_present(selected_formatter) then
```

Comment:
(Just a detail from a style nazi) I like to fail early instead of wrapping the whole code in the condition, it makes code easier to read since it reduces imbrications.
```
for loop_condition do
  if condition do
    code logic
  end
end

Becomes

for loop_condition do
  if not condition do
    return or continue
  end
  code logic
end
```

Reply:
i'm going through a couple of formatters and return the first available one
if none of theme where available, then i want to `return {}`
but updated this part a bit